src/burnchains/bitcoin/indexer.rs

```diff
@@ -472,22 +472,36 @@ impl BitcoinIndexer {
                     let interval_start_block =
                         (start_block / BLOCK_DIFFICULTY_CHUNK_SIZE).saturating_sub(2);
                     let base_block = interval_start_block * BLOCK_DIFFICULTY_CHUNK_SIZE;
-                    let interval_headers =
-                        canonical_spv_client.read_block_headers(base_block, start_block + 1)?;
-                    assert!(
-                        interval_headers.len() >= (start_block - base_block) as usize,
-                        "BUG: missing headers for {}-{}",
-                        base_block,
-                        start_block
-                    );
 
-                    test_debug!(
-                        "Copy headers {}-{}",
-                        base_block,
-                        base_block + interval_headers.len() as u64
-                    );
-                    reorg_spv_client
-                        .insert_block_headers_before(base_block - 1, interval_headers)?;
+                    if base_block > 0 {
+                        let interval_headers =
+                            canonical_spv_client.read_block_headers(base_block, start_block + 1)?;
+                        assert!(
+                            interval_headers.len() >= (start_block - base_block) as usize,
+                            "BUG: missing headers for {}-{}",
+                            base_block,
+                            start_block
+                        );
+
+                        debug!(
+                            "Copy headers {}-{}",
+                            base_block,
+                            base_block + interval_headers.len() as u64
+                        );
+                        reorg_spv_client
+                            .insert_block_headers_before(base_block - 1, interval_headers)?;
+                    } else {
+                        let interval_headers =
+                            canonical_spv_client.read_block_headers(1, start_block + 1)?;
+                        assert!(
+                            interval_headers.len() >= start_block as usize,
+                            "BUG: missing headers for 0-{}",
+                            start_block
+                        );
+
+                        debug!("Copy headers 0-{}", interval_headers.len() as u64);
```

Comment:
Should it be `Copy header 1-{}` since base block is manually set to 1 in this branch?

Reply:
Fixed